azkaban-common/src/main/java/azkaban/utils/AbstractMailer.java

```diff
@@ -53,24 +48,6 @@ public AbstractMailer(final Props props) {
 
     this.mailSender = props.getString("mail.sender", "");
     this.usesAuth = props.getBoolean("mail.useAuth", true);
-
-    this.clientHostname = props.get("server.hostname");
-    this.clientPort = props.getInt("server.port");
-    this.usesSSL = props.getBoolean("server.useSSL");
-
-    if (this.usesSSL) {
-      this.referenceURL =
-          "https://" + this.clientHostname
-              + (this.clientPort == 443 ? "/" : ":" + this.clientPort + "/");
-    } else {
-      this.referenceURL =
-          "http://" + this.clientHostname
-              + (this.clientPort == 80 ? "/" : ":" + this.clientPort + "/");
-    }
-  }
-
-  public String getReferenceURL() {
```

Comment:
Thanks for the comments. I want to remove this method also because it's using clientHostname, clientPortName and usesSSL, which are props that don't have default values. Alternative way would be to provide default value for all those three props, but since this getReferenceURL is not used anywhere in the code, I think it's safe to remove. I've tested it on localhost and it works properly.